packages/did/src/DidLinks/AccountLinks.chain.ts

```diff
@@ -0,0 +1,243 @@
+ * @param keyType The type of key/substrate account which produced the `signature`.
+ * @returns An [[Extrinsic]] that must be did-authorized.
+ */
+export async function getAccountSignedAssociationTx(
+  account: AccountAddress | AccountId,
+  signatureValidUntilBlock: AnyNumber,
+  signature: Uint8Array | HexString,
+  keyType: VerificationKeyType
+): Promise<Extrinsic> {
+  const blockchain = await BlockchainApiConnection.getConnectionOrConnect()
+  return blockchain.api.tx.didLookup.associateAccount(
+    account,
+    signatureValidUntilBlock,
+    { [keyType]: signature }
+  )
+}
+
+/**
+ * Returns a extrinsic to release an account link by the account that owns the deposit.
+ * Must be signed and submitted by the deposit owner account.
+ *
+ * @param linkedAccount Account whose link should be released (not the deposit owner).
+ * @returns The [[SubmittableExtrinsic]] for the `reclaimDeposit` call.
+ */
+export async function getReclaimDepositTx(
+  linkedAccount: AccountAddress | AccountId
+): Promise<SubmittableExtrinsic> {
+  const blockchain = await BlockchainApiConnection.getConnectionOrConnect()
+  return blockchain.api.tx.didLookup.reclaimDeposit(linkedAccount)
+}
+
+/**
+ * Allows the submitting account to unilaterally remove its link to a Did.
+ * This is not did-authorized, but directly submitted by the linked account.
+ *
+ * @returns A SubmittableExtrinsic that must be signed by the linked account.
+ */
+export async function getLinkRemovalByAccountTx(): Promise<SubmittableExtrinsic> {
+  const blockchain = await BlockchainApiConnection.getConnectionOrConnect()
+  return blockchain.api.tx.didLookup.removeSenderAssociation()
+}
+
+/**
+ * Allows the authorizing FullDid to unilaterally remove its link to a given account.
+ * This must be did-authorized, but can be submitted by any account.
+ *
+ * @param linkedAccount An account linked to the FullDid which should be unlinked.
+ * @returns An Extrinsic that must be did-authorized by the FullDid linked to `linkedAccount`.
+ */
+export async function getLinkRemovalByDidTx(
+  linkedAccount: AccountAddress | AccountId
+): Promise<Extrinsic> {
+  const blockchain = await BlockchainApiConnection.getConnectionOrConnect()
+  return blockchain.api.tx.didLookup.removeAccountAssociation(linkedAccount)
+}
+
+/* ### HIGH-LEVEL HELPERS ### */
+
+/**
+ * Builds an extrinsic to link `account` to `did`, where `account` must hold balance to cover for the transaction fees and deposit.
+ *
+ * @param account KeyringPair of the account to be linked, used for signing the did-authorized extrinsic.
+ * @param did FullDid to be linked, used for did-authorization.
+ * @param didSignCallback Keystore or callback that manages the FullDid's keys.
+ * @param signingOpts Options to be passed on to `FullDid.authorizeExtrinsic` & `Extrinsic.signAsync`.
+ * @returns A signed SubmittableExtrinsic ready to be submitted.
+ */
+export async function linkSubmitterAccount(
+  account: KeyringPair,
```

Comment:
I think we purposefully removed all instances of `KeyringPair`, since we most likely don't have access to the private key directly. Perhaps we should stick to that, and let the users handle the extrinsic the way they find best?

Reply:
I am not questioning its utility, but we don't have this type of function anywhere else in the chain files. It might make sense, but perhaps in a different file, if any.

Reply:
No, it already did very little (three lines essentially), so this is something that users can just put together themselves.